src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -0,0 +1,87 @@
+import json
+from binascii import unhexlify
+from random import sample
+
+from ipv8.community import Community
+from ipv8.lazy_community import lazy_wrapper
+from ipv8.messaging.lazy_payload import VariablePayload
+from ipv8.peer import Peer
+from ipv8.requestcache import RequestCache
+
+from tribler_core.modules.metadata_store.orm_bindings.channel_metadata import entries_to_chunk
+
+minimal_blob_size = 200
+maximum_payload_size = 1300
+max_entries = maximum_payload_size // minimal_blob_size
+max_query_peers = 5
+
+
+class RemoteSelectPayload(VariablePayload):
```

Comment:
Adding `@vp_compile` to your `VariablePayload` definitions makes them much faster. See also https://py-ipv8.readthedocs.io/en/latest/reference/serialization/ .